### PR TITLE
Check that selector exists before listening for change

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?

There was a bug in Frontend whereby some liveblogs and deadblogs failed to filter by key events on toggle. They would filter if the filter param was passed in as a query param. 

This bug was only replicable on football live/dead blogs. 

Example of pages where the filter doesn't work:
- https://www.theguardian.com/football/live/2021/nov/21/manchester-united-v-arsenal-womens-super-league-live
- https://www.theguardian.com/football/live/2021/nov/21/tottenham-v-leeds-united-premier-league-live-score-updates?filterKeyEvents=true#liveblog-content

The reason for this is that the enhanced football javascript is listening for an event change on `$('form.football-leagues')[0]` which assumes we always have this selector. We do not and so when we try to listen for a change on it, [Bean](https://github.com/fat/bean) - the event emitting library we use -  throws an error. This also explains why we only saw this bug on Football liveblogs as this JS is football specific.

This PR adds a check to see if we have this selector available before listening for changes to it. 

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### before
https://user-images.githubusercontent.com/20416599/155199980-8bb5d9f1-166c-4ea0-aaec-87a0131b8a39.mov

### after
https://user-images.githubusercontent.com/20416599/155199896-7dd964b3-70ec-4291-bd01-1e7974efd6a2.mov

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/football.js [hunk 1]
```diff
@@ -346,14 +346,17 @@ const init = () => {
...
-    bean.on(
-        document.body,
-        'change',
-        $('form.football-leagues')[0],
-        function onChange() {
-            window.location = this.value;
-        }
-    );
+
+    if ( $('form.football-leagues')[0]) {
```
Comment (joecowton1) [inline static/src/javascripts/bootstraps/enhanced/football.js:350]: I'd imagine you've thought of this but just to double check - could this ever cause an issue where we're looking for an index `[0]` of `undefined`?

Comment (marjisound) [reply static/src/javascripts/bootstraps/enhanced/football.js:350]: I also think it'd be safer if we check it has value `if ($('form.football-leagues') && $('form.football-leagues')[0])`

**Review by marjisound — APPROVED** [review]
Great work on finding where the root cause was 🥳

File: static/src/javascripts/bootstraps/enhanced/football.js (hunk 1 above)
Comment (abeddow91) [reply static/src/javascripts/bootstraps/enhanced/football.js:350]: Agreed, I'll amend

**Review by joecowton1 — APPROVED** [review]
LGTM

**Pull request merged**